Feedback method

eraydiler opened the pull request.

eraydiler committed:
Updated podfile to support newer cocoapods version, by removing
`exlusive` keyword.

eraydiler committed:
Removed storyboard support from project.

eraydiler committed:
Added a button to test feedback method.

eraydiler committed:
Added ability to show tryouts view controller modally when feedback
button tapped

eraydiler committed:
Added mockup icon and image for using with feedback view.

eraydiler committed:
Added purelayout to project.

eraydiler committed:
Made change to use autolayout

eraydiler committed:
Added folders for imagesets

eraydiler committed:
Added launch images to prevent wrong screen size issue

eraydiler committed:
Added pure layout into pods

eraydiler committed:
Added .bundle file to keep images inside of it, refactored code to use purelayout

eraydiler committed:
Added Settings.bundle

eraydiler committed:
Added background image and icon into example project’s imageset
temporarily.

eraydiler committed:
Added username text field

eraydiler committed:
Added PureLayout into podspec file, as required pod for the sdk.

eraydiler committed:
Removed PureLayout from Podfile.

eraydiler committed:
Added new overlayView class into project.

eraydiler committed:
Embedded view into navigation controller for.

eraydiler committed:
Added sample codes for feed back view.

eraydiler committed:
Removed PureLayout’s dependence from tryouts-SDK.

eraydiler committed:
Implemented auto layout using visual format language api.

eraydiler committed:
Added launch images.

eraydiler committed:
Added close button to feed back overlay view.

eraydiler committed:
Added delegate method for close button’s action.

eraydiler committed:
Added new protocol to inform user that feed back controller dismissed
after close button tapped.

eraydiler committed:
Orginized code for a clearer implementation.

eraydiler committed:
Made feed back view controller’s background transparent.

eraydiler committed:
Set layouts for shield and panel views.

eraydiler committed:
Removed lines commented before.

eraydiler committed:
Centered and resized feedback overlayview.

eraydiler committed:
Made background of feedback view transparent.

eraydiler committed:
Added image for the background of feedback overlay view.

eraydiler committed:
Fixed issue with untappable close button by setting
userInteractionEnabled property of its parent image view. Increased
tappable area of close button.

eraydiler committed:
Added comments to make code more understandable.

eraydiler committed:
Added username field.

eraydiler committed:
Added assets under Pod/Assests/ which was defined with s.resource_bundles… in the .podspec fie.

eraydiler committed:
Moved feedback overlay view’s background image into Assets folder in
development pods.

eraydiler committed:
Moved close button into Development pods/Assets directory.

eraydiler committed:
Added tryouts-icon into assets, reornised other images.

eraydiler committed:
Added ability to dismiss keyboard immeadiately when close button tapped.

eraydiler committed:
Added tryouts icon pinned on username tex field’s background image.

eraydiler committed:
Added constants to be able to change metrics easily.

eraydiler committed:
Reorganized code for a clearer implementation.

eraydiler committed:
Added images and prototype code for message view’s background

eraydiler committed:
Added submit button.

eraydiler committed:
Changed background image, started to change layout according to new
design prepared.

eraydiler committed:
Changed vertical offsets.

eraydiler committed:
Changed username field background image.

eraydiler committed:
Updated submit buttom’s background image.

eraydiler committed:
Readded assets folder and images to the project after pod install.

eraydiler committed:
Centered feedback button.

eraydiler committed:
Added message view centered into message view’s background with insets
of 5.0.

eraydiler committed:
Reorganized code to make visual format strings more readable.

eraydiler committed:
Added placeholder for the message view.

eraydiler committed:
Added delegate method for action of submit button.

eraydiler committed:
Added ability to get the screenshot for current window.

eraydiler committed:
Changed color and font of username field’s placeholder.

eraydiler committed:
Added private-method declerations.

eraydiler committed:
Made shieldView scroll view.

eraydiler committed:
Moved textview's delegate methods to feedback overlay class itself.

eraydiler committed:
Fixed merge conflicts

eraydiler committed:
Some fixes after merge.

eraydiler committed:
Added code for changing insets when keyboard showed or hid.

eraydiler committed:
Fixed a typo.

eraydiler committed:
Removed unuseful lines for cleaner code structure.

eraydiler committed:
Added content view as subview of scroll view to make layout scrollable.

eraydiler committed:
Added comments

eraydiler committed:
Moved shieldView’s code to up.

eraydiler committed:
Renamed example project classes, removed/added delegate methods for
`TRYFeedbackViewControlelr` added `TRYFeedback` model to hold data for
network request.

eraydiler committed:
Dismissed feedbackViewController when close button tapped, cleared
background colors.

eraydiler committed:
Sent feedback request and got response sucessfully.

eraydiler committed:
Added defensive check to prevent crash got when dictionary is <null>.

eraydiler committed:
Sent successful request with new credentials.

eraydiler committed:
Removed some code used for testing, commented lines, made cleanup.

taylanpince commented on Pod/Classes/Controllers/TRYFeedbackViewController.m:
```diff
@@ -0,0 +1,105 @@
+    UIGraphicsEndImageContext();
+
+    return screenshotImage;
+}
```
This is not the correct approach. Since iOS7 Apple provides a much faster and reliable approach for rendering views. Just look for examples using `snapshotViewAfterScreenUpdates:`

taylanpince commented on Pod/Classes/Controllers/TRYFeedbackViewController.m:
```diff
@@ -0,0 +1,105 @@
+
+#pragma mark - Screenshot
+
+- (UIImage *)screenShotOfWindow {
```
Let's please rename this method to `imageWithScreenshotOfActiveWindow`

taylanpince commented on Pod/Classes/Controllers/TRYFeedbackViewController.m:
```diff
@@ -0,0 +1,105 @@
+    return screenshotImage;
+}
+
+- (NSString *)encodedStringOfScreenshot {
```
Let's rename this to `stringWithEncodedScreenshotImageOfActiveWindow`

taylanpince commented on Pod/Classes/Tryouts.h:
```diff
@@ -48,4 +48,7 @@
                                   APIKey:(NSString *)APIKey
                                   secret:(NSString *)secret;
 
++ (void)presentFeedBackViewControllerFromViewController:(UIViewController *)viewController
```
Let's rename to `presentFeedbackControllerFromViewController:animated:`

taylanpince commented on Pod/Classes/Tryouts.m:
```diff
@@ -9,13 +9,16 @@
+#import "TRYFeedback.h"
 
 static NSString * const TRYAPIVersionCheckURL = @"https://api.tryouts.io/v1/applications/%@/";
+static NSString * const TRYAPIFeedbackSendURL = @"https://api-staging.tryouts.io/v1/applications/%@/feedback/";
```
Let's not forget to switch this to production before we merge

taylanpince commented on Pod/Classes/Views/TRYFeedbackOverlayView.m:
```diff
@@ -0,0 +1,633 @@
+     @"V:|-TRYOUTS_ICON_TOP-[tryoutsIcon(TRYOUTS_ICON_HEIGHT)]-ZERO_VERTICAL-[usernameBackground(USERNAME_BACKG_HEIGHT)]-MESSAGE_VIEW_BACKG_TOP-[messageBackground]-SUBMIT_BUTTON_TOP-[submitButton(SUBMIT_BUTTON_HEIGHT)]-SUBMIT_BUTTON_BOTTOM-|"
+                                            options:0
+                                            metrics:defaultMetrics
+                                              views:views]];
```
Holy shit. Good job, super clean.

taylanpince commented on Pod/Classes/Views/TRYFeedbackOverlayView.m:
```diff
@@ -0,0 +1,633 @@
+                                      resizingMode:UIImageResizingModeStretch]
+                            forState:UIControlStateNormal];
+
+    [submitButton setTitle:@"SUBMIT"
```
Needs to be localized

taylanpince commented on Pod/Classes/Views/TRYMessageView.m:
```diff
@@ -0,0 +1,76 @@
+    _placeholderLabel = [UILabel new];
+
+    _placeholderLabel.translatesAutoresizingMaskIntoConstraints = NO;
+    _placeholderLabel.text = @"Feedback...";
```
Needs to be localized

eraydiler committed:
Made changes for fixing issues with code review.

eraydiler committed:
Changed placeholder’s text.

eraydiler committed:
Added ability to save/get username in/from NSUserDefaults.

eraydiler committed:
Added new upload manager class, for dealing sending network request in
the background.

eraydiler committed:
Added feedback upload task model.

eraydiler committed:
Added code for implementing sending requests in the background.

eraydiler committed:
Made some corrections.

eraydiler committed:
Fixed issue with constant used for saving data into nsuserdefaults,
renamed constants, added break, and defensive check to terminate
unnecessary operations.

eraydiler committed:
Added missing fileds of uploadtask for serializing.

eraydiler committed:
Added ability to initialize upload manager when tryouts sdk is
initialized.

eraydiler committed:
Prevented upload manager from being called twice.

eraydiler committed:
Added missing fields.

eraydiler committed:
Fixed wrong key name

eraydiler committed:
Updated tasks in nsuserdefaults if all tasks are finished.

eraydiler committed:
Showd shake animation for invalid fields.

eraydiler committed:
Added ability to scroll automatically for which usernameField or
messageView is active.

eraydiler committed:
Removed unused property.

taylanpince commented on Pod/Classes/Views/TRYFeedbackOverlayView.m:
```diff
@@ -9,6 +9,9 @@
 #import "TRYMessageView.h"
 
+// NSUserDefaults
+static NSString const *kNSUserDefaulsUsernameKey = @"kNSUserDefaulsUsernameKey";
```
Let's make the actual string more unique. `com.hipo.tryouts.TRYUsername`

taylanpince commented on Pod/Classes/Network/TRYFeedbackUploadManager.m:
```diff
@@ -0,0 +1,107 @@
+static NSString * const kStorageDirectoryName        = @"kStorageDirectoryName";
+static NSString * const kTasksKey                    = @"kTasksKey";
+static NSString * const kBackgroundSessionIdentifier = @"kBackgroundSessionIdentifier";
+
```
Actual values should be more unique. Let's start them all with `com.hipo.tryouts.`

taylanpince commented on Pod/Classes/Network/TRYFeedbackUploadManager.h:
```diff
@@ -0,0 +1,19 @@
+
++ (TRYFeedbackUploadManager *)sharedManager;
+
+- (void)addFeedbackIntoTasks:(TRYFeedbackUploadTask *)feedback;
```
Rename to `createUploadTaskForFeedback:`

taylanpince commented on Pod/Classes/Tryouts.m:
```diff
@@ -206,4 +214,89 @@ - (void)alertView:(UIAlertView *)alertView clickedButtonAtIndex:(NSInteger)butto
+//                                  }];
+//
+//    [task resume];
+//}
```
Let's remove this?

eraydiler committed:
Removed commented lines.

eraydiler committed:
Made changes for code review.

eraydiler committed:
Dismissed keyboard immediately when submit button tapped.

eraydiler committed:
Added ability to dismiss keyboard if anywhere other than input fields
tapped.

eraydiler committed:
Made feedback view optional.

eraydiler committed:
Changed the way of getting screenshot, because the method implemented
before did not work in device.

eraydiler committed:
Addded clear button for username field.

eraydiler committed:
Added ability to change first responder as message view, when next
button is tapped in username field.

eraydiler committed:
Restored chages made for getting screenshot method.

eraydiler committed:
Restored back the way of taking screenshot of the active widow.

eraydiler committed:
Moved initializing feedback model and taking screenshot into init.

eraydiler committed:
Merge branch 'master' into feedback-method

eraydiler committed:
Updated cocoapods, and added missing line to podspec.

eraydiler committed:
Changed release version in podspec.

eraydiler committed:
Changed the name of folder from Assests to Assets.

eraydiler committed:
Added error log message.

eraydiler committed:
Made minor change to increase the quality of screenshot image.

eraydiler committed:
Refactored snapshot method with the correct implementation.

eraydiler committed:
Increased version in podspec by one.

eraydiler committed:
Increased close button’s tappable area.

eraydiler committed:
Changed transition style as cross dissolve, and set shield view’s
background color as black.

eraydiler committed:
Changed animation’s fade duration.

eraydiler committed:
Dismissed feedback controller smoother.

eraydiler committed:
Showed and dismissed panel view animated moving top.

eraydiler committed:
Increased font sizes for username field and message view.

eraydiler committed:
Fixed layout issue with username field, which was not centered while
clear button is active.

eraydiler committed:
Centered user name for some cases.

eraydiler committed:
Adjusted scroll view’s bottom inset and content size when keyboard
showed or dismissed.

eraydiler committed:
Fixed issue with scroll view’s content size which was calculated
incorrectly when the device orientation changed.

eraydiler committed:
Implemented custom clear button.

eraydiler committed:
Prevented issue with message view which accepts \n character as input
when next key is tapped from usernamefield.

eraydiler committed:
Made minor change to resign first responder when background tapped.

eraydiler committed:
Prevented keyboard to dismiss when submit button tapped with empty
username field.

eraydiler committed:
Removed unused constraint property, and a commented line.

eraydiler committed:
Added a minor fix to show moving up animation which occurs while feed
back overlay view is being displayed.

eraydiler committed:
Added ability to present feed back view when device is shaked.

eraydiler committed:
Opt outed a warning related to LaunchScreen.storyboard by checking
`requires full screen` option from target’s settings.

eraydiler committed:
Moved utility method into custom window’s class.

eraydiler committed:
Renamed property’s name.
